README.md

```diff
@@ -16,114 +23,119 @@ cd self_supervised
+For convenience, you can instead pass these parameters as keyword args, for example with `model = SelfSupervisedMethod(batch_size=128)`.
 
-To train a linear classifier on the result:
+## VICReg
+
+To train VICReg rather than MoCo v2, use the following parameters:
 
 ```python
+import os
 import pytorch_lightning as pl
-from linear_classifier import LinearClassifierMethod
-linear_model = LinearClassifierMethod.from_moco_checkpoint("example.ckpt")
-trainer = pl.Trainer(gpus=1, max_epochs=100)    
+from moco import SelfSupervisedMethod
+from model_params import VICRegParams
 
-trainer.fit(linear_model)
+os.environ["DATA_PATH"] = "~/data"
+
+params = VICRegParams()
+model = SelfSupervisedMethod(params)
+trainer = pl.Trainer(gpus=1, max_epochs=320)
+trainer.fit(model)
+trainer.save_checkpoint("example.ckpt")
 ```
 
+Note that we have not tuned these parameters for STL-10, and the parameters used for ImageNet are slightly different. See the comment on VICRegParams for details.
+
 ## BYOL
 
 To train BYOL rather than MoCo v2, use the following parameters:
 
 ```python
-from moco import MoCoMethodParams
-from moco import MoCoMethod
-params = MoCoMethodParams(
-    prediction_mlp_layers = 2,
-    mlp_normalization = "bn",
-    loss_type = "ip",
-    use_negative_examples_from_queue = False,
-    use_both_augmentations_as_queries = True,
-    use_momentum_schedule = True,
-    optimizer_name = "lars",
-    exclude_matching_parameters_from_lars = [".bias", ".bn"],
-    loss_constant_factor = 2
-)
-model = MoCoMethod(params)
-```
+import os
+import pytorch_lightning as pl
+from moco import SelfSupervisedMethod
+from model_params import BYOLParams
 
-For convenience, you can instead pass these parameters add keyword args, for example with `model = MoCoMethod(batch_size=128)`.
+os.environ["DATA_PATH"] = "~/data"
+
+params = BYOLParams()
+model = SelfSupervisedMethod(params)
+trainer = pl.Trainer(gpus=1, max_epochs=320)
+trainer.fit(model)
+trainer.save_checkpoint("example.ckpt")
+```
 
 ## SimCLR
 
 To run SimCLR, simply `use_negative_examples_from_batch` and disable `use_negative_examples_from_queue`. You can also set `K=0`: 
 
  ```python
-hparams = MoCoMethodParams(
-    use_negative_examples_from_batch=True,
-    use_negative_examples_from_queue=False,
-    K=0,
-    m=0.0,
-    use_both_augmentations_as_queries=True,
-)
+import os
+import pytorch_lightning as pl
+from moco import SelfSupervisedMethod
+from model_params import SimCLRParams
+
+os.environ["DATA_PATH"] = "~/data"
+
+params = SimCLRParams()
+model = SelfSupervisedMethod(params)
+trainer = pl.Trainer(gpus=1, max_epochs=320)
+trainer.fit(model)
+trainer.save_checkpoint("example.ckpt")
 ```
 
 This is not super efficient as it still has the separate offline network, and so it does each embedding calculation twice. However, it is sufficient for comparing the methods with each other.
```

Comment:
Can delete this line, we have no separate offline network for SimCLR now

Reply:
Done.